src/data/sqlGenerator.ts

```diff
@@ -149,9 +149,44 @@ const generateTraceIdQuery = (options: QueryBuilderOptions): string => {
   if (traceStatusCode !== undefined) {
     selectParts.push(`if(${escapeIdentifier(traceStatusCode.name)} IN ('Error', 'STATUS_CODE_ERROR'), 2, 0) as statusCode`);
   }
+
   const traceEventsPrefix = getColumnByHint(options, ColumnHint.TraceEventsPrefix);
   if (traceEventsPrefix !== undefined) {
-    selectParts.push(`arrayMap((name, timestamp, attributes) -> tuple(name, toString(toUnixTimestamp64Milli(timestamp)), arrayMap( key -> map('key', key, 'value', attributes[key]), mapKeys(attributes)))::Tuple(name String, timestamp String, fields Array(Map(String, String))),${escapeIdentifier(traceEventsPrefix.name)}.Name, ${escapeIdentifier(traceEventsPrefix.name)}.Timestamp, ${escapeIdentifier(traceEventsPrefix.name)}.Attributes) AS logs`);
```

Comment:
There are two ways in ClickHouse to define a Nested Table, controlled by the [`flatten_nested`](https://clickhouse.com/docs/en/operations/settings/settings#flatten-nested) setting at column creation time.

The query as previously written does not work when `flatten_nested=0`, as ClickHouse does not recognise the sub-columns such as `Events.Names` in this mode.

The rewritten query should work with either mode, as ClickHouse will recognise `Events` as one column even when it has been flattened.